src/libs/actions/Report.js

```diff
@@ -950,7 +951,7 @@ function handleReportChanged(report) {
  * @param {Number} reportID
  */
 function updateCurrentlyViewedReportID(reportID) {
-    Onyx.merge(ONYXKEYS.CURRENTLY_VIEWED_REPORTID, String(reportID));
+    Onyx.merge(ONYXKEYS.CURRENTLY_VIEWED_REPORTID, String(reportID || ''));
```

Comment:
I've updated this so it can be reused in `fetchAllReports` and to guard against `null` , `undefined` as `String(null)` would create a `"null"`  (similar situation for undefined and NaN)

Reply:
Hmm, this is the _one thing_ that I'm not so sure about. If we are calling this with a `reportID` that is `null` or `undefined` why try to set it at all? Couldn't we just leave whatever existing value we have? Seems not ideal to fail silently and set the `currentlyViewedReportID` to an empty string. Maybe it would be appropriate to log an alert of some kind when this happens instead?

We have some ways to do that e.g. 

https://github.com/Expensify/Expensify.cash/blob/f213be0a1be5a04599d2f17d8848b0a7824eb8be/src/libs/actions/Report.js#L717-L720

Reply:
> Hmm, this is the one thing that I'm not so sure about. If we are calling this with a reportID that is null or undefined why try to set it at all? 

As you can see the original logic would save anything without any consideration, I've just updated it so that if called with null or undefined they are converted

I'll just revert this, this will also address the used-before-defined problem

Reply:
Yes, I see. Only mentioning this since logging an alert would perhaps be useful. But no need to add it in this PR.